pkg/apis/cluster/types.go

```diff
@@ -104,6 +125,101 @@ type ClusterSpec struct {
 	// any resource that does not tolerate the Taint.
 	// +optional
 	Taints []corev1.Taint
+
+	// ResourceModels is the list of resource modeling in this cluster. Each modeling quota can be customized by the user.
+	// Modeling name must be one of the following: cpu, memory, storage, ephemeral-storage.
+	// If the user does not define the modeling name and modeling quota, it will be the default model.
+	// The default model grade from 0 to 8.
+	// When grade = 0 or grade = 1, the default model's cpu quota and memory quota is a fix value.
+	// When grade greater than or equal to 2, each default model's cpu quota is [2^(grade-1), 2^grade), 2 <= grade <= 7
+	// Each default model's memory quota is [2^(grade + 2), 2^(grade + 3)), 2 <= grade <= 7
+	// E.g. grade 0 likes this:
+	// - grade: 0
+	//   ranges:
+	//   - name: "cpu"
+	//     min: 0 C
+	//     max: 1 C
+	//   - name: "memory"
+	//     min: 0 GB
+	//     max: 4 GB
+	//
+	// - grade: 1
+	//   ranges:
+	//   - name: "cpu"
+	//     min: 1 C
+	//     max: 2 C
+	//   - name: "memory"
+	//     min: 4 GB
+	//     max: 16 GB
+	//
+	// - grade: 2
+	//   ranges:
+	//   - name: "cpu"
+	//     min: 2 C
+	//     max: 4 C
+	//   - name: "memory"
+	//     min: 16 GB
+	//     max: 32 GB
+	//
+	// - grade: 7
+	//   range:
+	//   - name: "cpu"
+	//     min: 64 C
+	//     max: 128 C
+	//   - name: "memory"
+	//     min: 512 GB
+	//     max: 1024 GB
+	//
+	// grade 8, the last one likes below. No matter what Max value you pass,
+	// the meaning of Max value in this grade is infinite. You can pass any number greater than Min value.
+	// - grade: 8
+	//   range:
+	//   - name: "cpu"
+	//     min: 128 C
+	//     max: MAXINT
+	//   - name: "memory"
+	//     min: 1024 GB
+	//     max: MAXINT
+	//
+	// +optional
+	ResourceModels []ResourceModel
+}
+
+// ResourceModel describes the modeling that you want to statistics.
+type ResourceModel struct {
+	// Grade is the index for the resource modeling.
+	// +optional
+	Grade int
```

Comment:
Nit: what's the expected behavior when user sets ResourceModels with `grade 1:CPU[2,4)`, `grade 3:CPU[8,16)`, `grade 4:CPU[16,32)`?

Reply:
BTW, IIRC we currently don't define the output of `kubectl describe cluster`, but it might be great if we can define a human readable format of ResourceModels like:


```
ResourceModels:
	Grade 1:	CPU [2,4)	Memory [4G,8G)
	Grade 2:	CPU [4,8)	Memory [8G,16G)
	Grade 3:	CPU [8,16)	Memory [16G,32G)
```

A method like `ResourceModel.Stringify()` shall be helpful in this feature.

Reply:
when user sets ResourceModels with `grade 1:CPU[2,4)`, `grade 3:CPU[8,16)`, `grade 4:CPU[16,32)`, if some resource's CPU is 6, this resource will be assigned to a higher grade. In other words, it will belong to `grade 3:CPU[8,16)`. Does it make sense? 

By the way, in this case, grade2 doesn't exist? Or the user forgot to enter it?

Reply:
This `ResourceModel.Stringify()` feature is very nice. If time is enough, I can do it together in this version.

Reply:
> when user sets ResourceModels with grade 1:CPU[2,4), grade 3:CPU[8,16), grade 4:CPU[16,32), if some resource's CPU is 6, this resource will be assigned to a higher grade.

+1. We can take this as the default behavior. But I don't think we should note this case.

> By the way, in this case, grade2 doesn't exist? Or the user forgot to enter it?

This is an abnormal case, probably due to a mistake. 
This usage is ambiguous, my suggestion is to validate the input at webhook(Validation Plugin) to avoid discontinuous ranges.